phoenix-core-client/src/main/java/org/apache/phoenix/compile/QueryCompiler.java

```diff
@@ -705,6 +717,9 @@ protected QueryPlan compileSubquery(
                     statement,
                     queryPlan);
         }
+        if (parentContext != null) {
+            parentContext.addSubStatementContext(queryPlan.getContext());
```

Comment:
What is the reasoning behind this change?

Reply:
@tkhurana This is to keep track of the Context for nested/inner Select queries. During `getReadMetrics` in `PhoenixResultSet`, we combine the metrics from the inner contexts with the outer context.